workflow/scripts/add_extra_components.py

```diff
@@ -882,6 +883,346 @@ def trim_network(n, trim_topology):
     n.determine_network_topology()
 
 
+def add_co2_storage(n: pypsa.Network, config: dict, co2_storage_csv: str, costs: pd.DataFrame, sector: bool):
+    """Adds node level CO2 (underground) storage."""
+    # get node level CO2 (underground) storage potential and cost from CSV file
+    co2_storage = pd.read_csv(co2_storage_csv).set_index("node")
+
+    # add buses to represent node level CO2 captured by different processes
+    n.madd(
+        "Bus",
+        co2_storage.index,
+        suffix=" co2 capture",
+        carrier="co2",
+    )
+
+    # add stores to represent node level CO2 (underground) storage
+    n.madd(
+        "Store",
+        co2_storage.index,
+        suffix=" co2 storage",
+        bus=co2_storage.index + " co2 capture",
+        e_nom_extendable=True,
+        e_nom_max=co2_storage["potential [MtCO2]"] * 1e6,  # in tCO2
+        marginal_cost=co2_storage["cost [USD/tCO2]"],
+        carrier="co2",
+    )
+
+    # add carrier to represent CO2
+    n.madd(
+        "Carrier",
+        ["co2"],
+        color=config["plotting"]["tech_colors"]["co2"],
+        nice_name=config["plotting"]["nice_names"]["co2"],
+    )
+
+    # add carrier to represent CC only (i.e. without S)
+    carriers = n.carriers.query("Carrier.str.endswith('CCS')")
+    if carriers.empty is False:
+        n.madd(
+            "Carrier",
+            carriers.index.str.replace("CCS", "CC", regex=True),
+            color=carriers["color"],
+            nice_name=carriers["nice_name"].str.replace("Ccs", "Cc", regex=True),
+        )
+
+    # get CO2 intensity for gas and coal
+    gas_co2_intensity = costs.loc["gas"]["co2_emissions"]
+    coal_co2_intensity = costs.loc["coal"]["co2_emissions"]
+
+    if sector is True:
+        links = n.links.index.str.contains("CCS")
+        if True in links:  # found links equipped with CCS
+            # specify links' bus4 to point to their respective CO2 capture buses
+            n.links.loc[links, "bus4"] = co2_storage.index + " co2 capture"
+
+            # calculate efficiencies
+            efficiency2 = []
+            efficiency4 = []
```

Comment:
This is my fault for not documenting the different link destinations. But can we just add in comments here on the destination? (If my comment below is correct?) 

```suggestion
            efficiency2 = [] # to atmosphere
            efficiency4 = [] # to underground storage
```

Reply:
This review/feedback is solved/implemented with commit https://github.com/ricnogfer/pypsa-usa/commit/319d04b61b60363ba06437840699ca8023ef9a99.